tools/codesign_macOS_executable/sign_executable.py

```diff
@@ -0,0 +1,177 @@
+                "-rpath",
+                "/Volumes/Work/Perforce/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac",
+                os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac"),
+                dylib]
+            print(f"[SPEAR | sign_executable.py] Executing: {' '.join(cmd)}")
+            cmd_result = subprocess.run(cmd)
+            assert cmd_result.returncode == 0
+
+        # revert current working directory
+        print(f"[SPEAR | sign_executable.py] Changing working directory to {cwd}.")
+        os.chdir(cwd)
+
+        # files that need to be codesigned
+        sign_files = [
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac", "libogg.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac", "libvorbis.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPhysX3.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPhysX3Common.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPxFoundation.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libAPEX_Clothing.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPxPvdSDK.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libNvCloth.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPhysX3Cooking.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libAPEX_Legacy.dylib"),
+            os.path.join(executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libApexFramework.dylib"),
+            os.path.join(executable, "Contents", "MacOS", os.path.splitext(executable_name)[0])]
+
+        assert os.path.exists(args.entitlements_file)
+
+        for file in sign_files:
+            cmd = [
+                "sudo", "codesign", "-f", "-s", "-v", "--options", "runtime", "--timestamp", "--entitlements", 
+                args.entitlements_file, "--sign", f"Developer ID Application: {args.developer_id}", file]
+            print(f"[SPEAR | sign_executable.py] Executing: {' '.join(cmd)}")
+            cmd_result = subprocess.run(cmd)
+            assert cmd_result.returncode == 0
+
+        os.makedirs(args.output_dir, exist_ok=True)
+        os.makedirs(args.tmp_dir, exist_ok=True)
+
+        # create a zip file for notarization
+        notrn_zip = os.path.join(args.tmp_dir, f"{os.path.splitext(os.path.basename(executable))[0]}.zip")
```

Comment:
notarization_zip

Reply:
You already assign os.path.basename(executable) to another variable, so use the variable you created of repeating os.path.basename(executable)

Reply:
Done